### DISCUSSION TITLE
How can I tell if a deep link was successfully matched to a screen or not?

### DISCUSSION THREAD

**Discussion by itsramiel** [react-navigation/react-navigation#12068] [Q&A]
Hi,

I have my react native app with react navigation setup for deep linking as mentioned in the [deep linking docs](https://reactnavigation.org/docs/7.x/deep-linking/).

But what I would like to know is that when a user opens the app with a deep link, whether that deep link successfully navigated the user to a screen or not. As in is the deep link a valid link for a screen or not?

How can I figure that out?

**Comment by chapl** [comment]
I handle links in a hook, decode the url and process it with a switch statement. If it doesn't match, it'll hit the default block.

**Reply by itsramiel** [reply]
Thanks for the idea

**Comment by satya164** [comment]
You can use the `getStateFromPath` function. It'll return a state object if there's a match:

```js
getStateFromPath(path, config) != null
```

**Reply by itsramiel** [reply]
Alright, thanks will check if that works for conditionally rendered paths.
